### PR TITLE
feat(animate-api): add support for exit animation

### PR DESCRIPTION
**Pull request by sidMulay**
### JIRA Ticket

Jira:  [BB-7912](https://hello-haptik.atlassian.net/browse/BB-7912)

---

### Changelog

- Add support for exit animations in Animate API
- Fix export
- Refactor to utilize variants from styled system

Test link: https://sid-relements.netlify.app/?path=/story/design-system-helpers-animate--documentation

### CODE REVIEW

File: src/components/Animate/animate.storybook.js [hunk 1]
```diff
@@ -35,147 +38,274 @@ SyntaxHighlighter.registerLanguage("jsx", jsx);
...
+          {importCode}
+        </SyntaxHighlighter>
+        <br />
+
+        <Text variant="h2.semi-bold">Configuration</Text>
+        <br />
+        <ConfigBlock of={Animate} />
+
+        <Text variant="h2.semi-bold">Currently supported transitions</Text>
+        <Text>{renderAnimations()}</Text>
```
Comment (sidMulay) [inline src/components/Animate/animate.storybook.js:81]: This design is to be improved, just kept here as a placeholder.

File: config/exports.js [hunk 2]
```diff
@@ -51,4 +51,5 @@ export default {
   'decorators/activify': 'src/decorators/Activify/index.js',
 
   'theme/provider': 'src/components/Context/Provider.js',
+  'components/animate': 'src/components/Animate/index.js',
```
Comment (gopal-virtual) [inline config/exports.js:54]: keep it under `Layout`

File: src/Theme/animateAPI.js [hunk 3]
```diff
@@ -42,6 +54,39 @@ export const animate = {
       }
     }
 
+    if (isExiting) {
+      const exitDirection = getDirection(exit);
+
+      if (exitDirection) {
+        if (
```
Comment (gopal-virtual) [inline src/Theme/animateAPI.js:61]: if/else is not scalable, use a map instead

**Review by gopal-virtual — CHANGES_REQUESTED** [review]

**Review by yashasvi13 — APPROVED** [review]

**Pull request merged**